docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,430 @@
+      resourceConfig: {
+        instructionLeeway: 50000,
+      },
+    },
+  };
+
+  const response = await fetch("https://soroban-testnet.stellar.org:443", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(requestBody),
+  });
+
+  const result = await response.json();
+  console.log(JSON.parse(result));
+}
+
+// Example XDR transaction envelope
+// Replace the following placeholder with your actual XDR transaction envelope
+const transactionXDR =
+  "AAAAAgAAAAAg4dbAxsGAGICfBG3iT2cKGYQ6hK4sJWzZ6or1C5v6GAAAAGQAJsOiAAAAEQAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAGAAAAAAAAAABzAP+dP0PsNzYvFF1pv7a8RQXwH5eg3uZBbbWjE9PwAsAAAAJaW5jcmVtZW50AAAAAAAAAgAAABIAAAAAAAAAACDh1sDGwYAYgJ8EbeJPZwoZhDqEriwlbNnqivULm/oYAAAAAwAAAAMAAAAAAAAAAAAAAAA=";
+// An example of where to get the XDR is from TransactionBuilder class from the sdk as shown in the next example.
+simulateTransaction(transactionXDR);
+```
+
+### Using the JavaScript SDK
+
+The Stellar SDK provides a convenient method to simulate a transaction:
+
+```javascript
+import {
+  Keypair,
+  SorobanRpc,
+  scValToNative,
+  TransactionBuilder,
+  BASE_FEE,
+  Networks,
+  Operation,
+} from "@stellar/stellar-sdk";
+
+const FRIENDBOT_URL = "https://friendbot-testnet.stellar.org/";
+
+const rpc_url = "https://soroban-testnet.stellar.org:443";
+
+// Generate a new keypair for transaction authorization.
+const keypair = Keypair.random();
+const secret = keypair.secret();
+const publicKey = keypair.publicKey();
+console.log("publicKey:", publicKey);
+// you need to fund the account.
+await fetch(`https://friendbot-testnet.stellar.org/?addr=${publicKey}`).then(
+  (res) => {
+    console.log(`funded account: ${publicKey}`);
+  },
+);
+
+// Initialize the rpcServer
+const RpcServer = new SorobanRpc.Server(rpc_url, { allowHttp: true });
+
+// Load the account (getting the sequence number for the account and making an account object.)
+const account = await RpcServer.getAccount(publicKey);
+
+// Define the transaction
+const transaction = new TransactionBuilder(account, {
+  fee: BASE_FEE,
+})
+  .setNetworkPassphrase(Networks.TESTNET)
+  .setTimeout(30)
+  .addOperation(
+    Operation.invokeContractFunction({
+      function: "symbol",
+      // the contract function and address need to be set by you.
+      contract: "CDLZFC3SYJYDZT7K67VZ75HPJVIEUVNIXF47ZG2FB2RMQQVU2HHGCYSC",
+      args: [],
+    }),
+  )
+  .build();
+// If you want to get this as an XDR string directly, you would use `transaction.toXDR('base64')`
+
+RpcServer.simulateTransaction(transaction).then((sim) => {
+  console.log("cost:", sim.cost);
+  console.log("result:", sim.result);
+  // the result is a ScVal and so we can parse that to human readable output using the sdk's `scValToNative` function:
+  console.log("humanReadable Result:", scValToNative(sim.result?.retval));
+  console.log("error:", sim.error);
+  console.log("latestLedger:", sim.latestLedger);
+});
+```
+
```

Comment:
```suggestion
```
To run the above code, you will need to install the Stellar SDK into your project. You can do this by running the following command in your project directory:

```
npm install @stellar/stellar-sdk
``` 



Once your project is set up, you can create a new `mjs` file and paste the code above. You can then run the file using Node.js by running:

```
node <filename>.mjs
```

Reply:
> To run the above code, you will need to install the Stellar SDK into your project. You can do this by running the following command in your project directory:
> 
> ```
> npm install @stellar/stellar-sdk
> ```
> 
> Once your project is set up, you can create a new `mjs` file and paste the code above. You can then run the file using Node.js by running:
> 
> ```
> node <filename>.mjs
> ```

added these instructions at line 170.  It' can finally be merged.